src/app/static/src/app/root.ts

```diff
@@ -42,6 +47,34 @@ const persistState = (store: Store<RootState>) => {
     })
 };
 
+const resetState = (store: Store<RootState>) => {
+    store.subscribe((mutation: MutationPayload, state: RootState) => {
+
+        if (state.baseline.ready
+            && state.surveyAndProgram.ready
+            && state.modelRun.ready) {
+
+            const type = stripNamespace(mutation.type);
+
+            if (BaselineUpdates.includes(type as BaselineMutation)) {
+                store.commit("ResetInputs");
```

Comment:
In fact this `ResetInputs` is just resetting the `filteredData` state (the actions handle all the deletion of input files.) So perhaps this should be renamed.

Reply:
Ah so no, I've only moved control for committing __mutations__ to the plugin, not for dispatching __actions__. It's not 100% clear to me from the Vuex docs, but I was under the impression plugins should only commit mutations, not dispatch actions. They don't make this very explicit, but they describe a plugin as:

"The plugin simply receives the store as the only argument and can either listen to mutations (for outbound data persistence, logging, or debugging) or dispatch mutations (for inbound data e.g. websockets or observables)." (https://vuex.vuejs.org/api/#plugins)

Reply:
Just noticed that confusingly there they say "dispatch" mutations. Lol.